### PR TITLE
FPASF-460: Parallelise copilot deployments

### PR DESCRIPTION
**Pull request by MarcUsher**
### Change description
Updates the GitHub deployment workflow to run copilot deploys of `celery` & `post-award` services and the `download-report` job in parallel. 

See example deployment run: https://github.com/communitiesuk/funding-service-design-post-award-data-store/actions/runs/10176167218

- [ ] Unit tests and other appropriate tests added or updated
- [ ] README and other documentation has been updated / added (if needed)
- [X] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

### How to test
_If manual testing is needed, give suggested testing steps_

### CODE REVIEW

File: .github/workflows/deploy_combined_service.yml [hunk 1]
```diff
@@ -34,26 +38,29 @@ jobs:
...
       owner: ${{ github.repository_owner }}
       application: ${{ github.event.repository.name }}
-  deployment:
-    concurrency: deploy-${{ inputs.environment || 'test' }}
+
+  copilot_deploy:
+    needs: [ tag_version, paketo_build ]
+    strategy:
+      matrix:
+        deployments: [ post-award, post-award-celery, download-report ]
```
Comment (samuelhwilliams) [inline .github/workflows/deploy_combined_service.yml:46]: Just flagging a potential alternative - not saying one way or the other is better. You can decide if you like it or not.

---

If we used the `include` matrix key we could also add a `copilot_command` parameter (or something) to feed the `copilot` command later.

Eg

```suggestion
    strategy:
      matrix:
        include:
          - deployment: post-award
            command: svc
          - deployment: post-award-celery
            command: svc
          - deployment: download-report
            command: job
```

And then at the bottom we could have:

```
      - name: Copilot deploy ${{ matrix.deployments }}
        run: |
          copilot ${{ matrix.command }} deploy --name ${{ matrix.deployments }}
```

Not convinced (or unconvinced) which is better, so you make a call.

Comment (MarcUsher) [reply .github/workflows/deploy_combined_service.yml:46]: Oh yeah this is a lot nicer, makes it much more clear and readable. Nice suggestion, will add this now!

Comment (MarcUsher) [reply .github/workflows/deploy_combined_service.yml]: @samuelhwilliams have added this suggestion to the PR, thanks!

**Review by samuelhwilliams — APPROVED** [review]
noice ✅

**Pull request merged**